staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_test.go

```diff
@@ -215,6 +215,48 @@ func TestParseQuantityString(t *testing.T) {
 	}
 }
 
+func TestQuantityToInt64(t *testing.T) {
```

Comment:
I'd like a test that uses maxInt64Factors directly in the constant value (such that we show that `N > expr(maxInt64Factors)` uses slow path, `N <= expr(maxInt64Factors)` uses fast path, and that fastPath is shown not to overflow when given a number derived from `maxInt64Factors`.

Reply:
> I'd like a test that uses maxInt64Factors directly in the constant value (such that we show that `N > expr(maxInt64Factors)` uses slow path, `N <= expr(maxInt64Factors)` uses fast path, and that fastPath is shown not to overflow when given a number derived from `maxInt64Factors`.

N(binarySI) <= expr(maxInt64Factors) may also use slow path, for example 1000000000Gi in the test is shown overflow, we can only ensure fast path num not overflow.
```
// 1Mi (2^20) has ~6 digits of decimal precision, so exponent*3/10 -1 is roughly the precision
precision = maxInt64Factors - int32(len(num)) - int32(float32(exponent)*3/10)
```